### ISSUE TITLE
Distribute patched headers (CCCL, fmt, spdlog, etc.) in `rapids-core-dependencies`

### ISSUE THREAD

**Issue by jakirkham** [rapidsai/build-planning#109]
> I believe https://github.com/rapidsai/raft/pull/2284 fixed the issues with `raft` vendoring `rmm` headers. Think the right long-term solution for the CCCL + `fmt` + `spdlog` stuff is to distribute `rapids-core-dependencies` with those things in it, which I'd love to help with (cc @robertmaynard)  but probably couldn't get done before we enter burndown for 24.06. 

 _Originally posted by @jameslamb in [#54](https://github.com/rapidsai/build-planning/issues/54#issuecomment-2093814987)_

Note: [rapids-cmake]( https://github.com/rapidsai/rapids-cmake ) contains the recipe for [`rapids_core_dependencies`]( https://github.com/rapidsai/rapids-cmake/tree/main/conda/recipes/rapids_core_dependencies )

**Comment by vyasr** [comment]
I do not believe that this is desirable. We should move away from shipping patched dependencies. Vendoring dependencies that are part of our public API is a recipe for subtle ABI compatibility issues that we should avoid. With https://github.com/rapidsai/build-planning/issues/104 closing, we have taken a major step forward by removing fmt and spdlog from the equation. We are actively working with the CCCL team to get RAPIDS building against the latest trunk of that repository. We should prioritize that effort so that we can stop needing to use patched versions of CCCL and rely solely on published packages. Once that is done we will have addressed the three main reasons that we have needed to ship dependencies in the past.
